### PR TITLE
Implement ccipChainReader.LinkPriceUSD()

### PR DESCRIPTION
**Pull request by rstout**
Implement a function in ccipChainReader to get the price of the LINK token. We first call the dest chain FeeQuoter's `GetStaticConfig` method to get the address of the LINK token, then we call the `GetTokenPrices` method to get the price of LINK.

### CODE REVIEW

File: pkg/reader/ccip.go [hunk 1]
```diff
@@ -775,6 +775,81 @@ func (r *ccipChainReader) GetContractAddress(contractName string, chain cciptype
 	return addressBytes, nil
 }
 
+// GetLinkPriceUSD gets the LINK price in USD from the FeeQuoter contract on the destination chain.
+func (r *ccipChainReader) GetLinkPriceUSD(ctx context.Context) (cciptypes.BigInt, error) {
+	// Exit without an error if we cannot read the destination.
```
Comment (makramkd) [inline pkg/reader/ccip.go:780]: This comment contradicts the code below - do we wanna exit w/ and error or without?

File: pkg/reader/ccip.go [hunk 2]
```diff
@@ -775,6 +775,81 @@ func (r *ccipChainReader) GetContractAddress(contractName string, chain cciptype
 	return addressBytes, nil
 }
 
+// GetLinkPriceUSD gets the LINK price in USD from the FeeQuoter contract on the destination chain.
+func (r *ccipChainReader) GetLinkPriceUSD(ctx context.Context) (cciptypes.BigInt, error) {
```
Comment (makramkd) [inline pkg/reader/ccip.go:779]: nit: call this just `LinkPriceUSD`?

File: pkg/reader/ccip.go [hunk 3]
```diff
@@ -775,6 +775,81 @@ func (r *ccipChainReader) GetContractAddress(contractName string, chain cciptype
...
+	// Exit without an error if we cannot read the destination.
+	if err := validateExtendedReaderExistence(r.contractReaders, r.destChain); err != nil {
+		return cciptypes.BigInt{}, fmt.Errorf("validate extended reader existence: %w", err)
+	}
+
+	feeQuoterCfg, err := r.getDestFeeQuoterStaticConfig(ctx)
```
Comment (makramkd) [inline pkg/reader/ccip.go:785]: Is this cache-able? Don't need to cache the static configs now, but can put a TODO for later perhaps.

File: pkg/reader/ccip.go [hunk 4]
```diff
@@ -775,6 +775,81 @@ func (r *ccipChainReader) GetContractAddress(contractName string, chain cciptype
...
+	if err != nil {
+		return cciptypes.BigInt{}, fmt.Errorf("get destination fee quoter static config: %w", err)
+	}
+
+	linkTokenAddr := feeQuoterCfg.LinkToken
+
+	getLinkPriceUSD, err := r.getLinkPriceUSD(ctx, linkTokenAddr)
+
+	if err != nil {
```
Comment (makramkd) [inline pkg/reader/ccip.go:794]: Inline and reduce newlines

```suggestion
	getLinkPriceUSD, err := r.getLinkPriceUSD(ctx, feeQuoterCfg.LinkToken)
	if err != nil {
```

File: pkg/reader/ccip.go [hunk 5]
```diff
@@ -775,6 +775,81 @@ func (r *ccipChainReader) GetContractAddress(contractName string, chain cciptype
...
+	err := r.getDestinationData(
+		ctx,
+		r.destChain,
+		consts.ContractNameFeeQuoter,
+		consts.MethodNameFeeQuoterGetStaticConfig,
+		&staticConfig,
+	)
+
+	if err != nil {
+		return feeQuoterStaticConfig{}, fmt.Errorf("unable to lookup nonce manager (offramp static config): %w", err)
```
Comment (makramkd) [inline pkg/reader/ccip.go:813]:
```suggestion
		return feeQuoterStaticConfig{}, fmt.Errorf("unable to lookup fee quoter static config: %w", err)
```

File: pkg/reader/ccip.go [hunk 6]
```diff
@@ -775,6 +775,81 @@ func (r *ccipChainReader) GetContractAddress(contractName string, chain cciptype
...
+// TODO: doc
+func (r *ccipChainReader) getLinkPriceUSD(ctx context.Context, linkTokenAddr []byte) (cciptypes.BigInt, error) {
+	if len(linkTokenAddr) == 0 {
+		return cciptypes.BigInt{}, fmt.Errorf("linkTokenAddr is empty")
+	}
+
+	reader, ok := r.contractReaders[r.destChain]
+	if !ok {
+		return cciptypes.BigInt{}, fmt.Errorf("contract reader not found for chain %d", r.destChain)
+	}
```
Comment (makramkd) [inline pkg/reader/ccip.go:828]: This is the second time this check is done in this call stack, dedup?

File: pkg/reader/ccip.go [hunk 7]
```diff
@@ -775,6 +775,81 @@ func (r *ccipChainReader) GetContractAddress(contractName string, chain cciptype
 	return addressBytes, nil
 }
 
+// GetLinkPriceUSD gets the LINK price in USD from the FeeQuoter contract on the destination chain.
```
Comment (makramkd) [inline pkg/reader/ccip.go:778]: Maybe state the denomination here (i.e how many decimals)

File: pkg/reader/ccip.go (hunk 6 above)
Comment (rstout) [reply pkg/reader/ccip.go:828]: The alternative is to get rid of the `validateExtendedReaderExistence` call, but that's a stronger check. I'm fine with leaving both in.

File: pkg/reader/ccip.go (hunk 7 above)
Comment (rstout) [reply pkg/reader/ccip.go:778]: Aside: these implicit decimals/denominations are very confusing imo

File: pkg/reader/ccip.go (hunk 6 above)
Comment (asoliman92) [reply pkg/reader/ccip.go:826]: This function is a duplicate from `OnchainTokenPricesReader.GetFeeQuoterTokenUpdates`. Maybe pass the price reader to CCIP and use it instead.

**Review by makramkd — APPROVED** [review]

File: pkg/reader/ccip.go (hunk 6 above)
Comment (rstout) [reply pkg/reader/ccip.go:826]: `GetWrappedNativeTokenPriceUSD` also makes the same FeeQuoter call, and there's other FeeQuoter calls in `ccipChainReader`. I think it makes sense to consolidate FeeQuoter functionality in a future PR, "after the dust settles".

**Pull request merged**